Sources/ArcGISToolkit/Components/Authentication/CertificatePickerViewModifier.swift

```diff
@@ -92,11 +103,41 @@ import ArcGIS
         challenge.resume(with: .cancel)
     }
 
-    private func showCertificateImportError(_ error: Error?) async {
+    private func showCertificateError(_ error: CertificateError) async {
         // This is required to prevent an "already presenting" error.
         try? await Task.sleep(nanoseconds: 100_000)
-        certificateImportError = error as? CertificateImportError
-        showCertificateImportError = true
+        certificateError = error
+        showCertificateError = true
+    }
+}
+
+extension CertificateImportError: LocalizedError {
+    public var errorDescription: String? {
+        switch self {
+        case .invalidData:
+            return String(localized: "The certificate file was invalid.", bundle: .module)
+        case .invalidPassword:
+            return String(localized: "The password was invalid.", bundle: .module)
+        default:
+            let errorString = SecCopyErrorMessageString(rawValue, nil) as? String ?? String(
+                localized: "The certificate file or password was invalid.",
+                bundle: .module
+            )
+            return errorString
```

Comment:
Since `CFString` is bridgeable to `String`, the cast doesn't need to be conditional.

```suggestion
            return SecCopyErrorMessageString(rawValue, nil) as String? ?? String(
                localized: "The certificate file or password was invalid.",
                bundle: .module
            )
```

Reply:
done